python/uv/private/versions.bzl

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Version and integrity information for downloaded artifacts"""
+
+UV_PLATFORMS = {
+    "aarch64-apple-darwin": struct(
+        default_repo_name = "rules_python_uv_darwin_aarch64",
+        compatible_with = [
+            "@platforms//os:macos",
+            "@platforms//cpu:aarch64",
+        ],
+    ),
+    "aarch64-unknown-linux-gnu": struct(
+        default_repo_name = "rules_python_uv_linux_aarch64",
+        compatible_with = [
+            "@platforms//os:linux",
+            "@platforms//cpu:aarch64",
+        ],
+    ),
+    "powerpc64le-unknown-linux-gnu": struct(
+        default_repo_name = "rules_python_uv_linux_ppc",
+        compatible_with = [
+            "@platforms//os:linux",
+            "@platforms//cpu:ppc",
+        ],
+    ),
+    "s390x-unknown-linux-gnu": struct(
+        default_repo_name = "rules_python_uv_linux_s390x",
+        compatible_with = [
+            "@platforms//os:linux",
+            "@platforms//cpu:s390x",
+        ],
+    ),
+    "x86_64-apple-darwin": struct(
+        default_repo_name = "rules_python_uv_darwin_x86_64",
+        compatible_with = [
+            "@platforms//os:macos",
+            "@platforms//cpu:x86_64",
+        ],
+    ),
+    "x86_64-pc-windows-msvc": struct(
+        default_repo_name = "rules_python_uv_windows_x86_64",
+        compatible_with = [
+            "@platforms//os:windows",
+            "@platforms//cpu:x86_64",
+        ],
+    ),
+    "x86_64-unknown-linux-gnu": struct(
+        default_repo_name = "rules_python_uv_linux_x86_64",
+        compatible_with = [
+            "@platforms//os:linux",
+            "@platforms//cpu:x86_64",
+        ],
+    ),
+}
+
+# From: https://github.com/astral-sh/uv/releases
+UV_TOOL_VERSIONS = {
```

Comment:
FYI @groodt you may be interested in https://github.com/aspect-build/rules_lint/blob/main/lint/mirror_ruff.sh which auto-updates our mirror of Ruff, another tool distributed in the same way by astral-sh. We have GitHub Actions send the update PRs so no humans need to do anything.